2025-06-11 07:17:45 +0000 - dagster - DEBUG - __ASSET_JOB - c6d0ef2b-7758-4f37-80ac-1ca033191a6c - 1163 - LOGS_CAPTURED - Started capturing logs in process (pid: 1163).
2025-06-11 07:17:45 +0000 - dagster - DEBUG - __ASSET_JOB - c6d0ef2b-7758-4f37-80ac-1ca033191a6c - 1163 - get_monthly_lotto_result - STEP_START - Started execution of step "get_monthly_lotto_result".
2025-06-11 07:17:45,191 - INFO - Context impl SQLiteImpl.
2025-06-11 07:17:45,191 - INFO - Will assume non-transactional DDL.
2025-06-11 07:17:45,373 - INFO - Starting get_monthly_lotto_result asset execution.
2025-06-11 07:17:45,742 - INFO - Launching Chromium browser to scrape results for May 2025 to May 2025.
2025-06-11 07:17:46,760 - INFO - Navigated to PCSO Search Lotto Result page.
2025-06-11 07:17:46,760 - INFO - Selecting start date: May 01, 2025
2025-06-11 07:18:16,907 - ERROR - An unexpected error occurred during Playwright operations: ElementHandle.select_option: Timeout 30000ms exceeded.
Call log:
  - attempting select option action
    2 × waiting for element to be visible and enabled
      - did not find some options
    - retrying select option action
    - waiting 20ms
    2 × waiting for element to be visible and enabled
      - did not find some options
    - retrying select option action
      - waiting 100ms
    59 × waiting for element to be visible and enabled
       - did not find some options
     - retrying select option action
       - waiting 500ms
Traceback (most recent call last):
  File "/app/src/lotto_project/assets/lotto_result.py", line 124, in get_monthly_lotto_result
    page.wait_for_selector("id=cphContainer_cpContent_ddlStartDate").select_option(value=day_from)
  File "/app/.venv/lib/python3.12/site-packages/playwright/sync_api/_generated.py", line 2291, in select_option
    self._sync(
  File "/app/.venv/lib/python3.12/site-packages/playwright/_impl/_sync_base.py", line 115, in _sync
    return task.result()
           ^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/playwright/_impl/_element_handle.py", line 164, in select_option
    return await self._channel.send("selectOption", params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/playwright/_impl/_connection.py", line 61, in send
    return await self._connection.wrap_api_call(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/playwright/_impl/_connection.py", line 528, in wrap_api_call
    raise rewrite_error(error, f"{parsed_st['apiName']}: {error}") from None
playwright._impl._errors.TimeoutError: ElementHandle.select_option: Timeout 30000ms exceeded.
Call log:
  - attempting select option action
    2 × waiting for element to be visible and enabled
      - did not find some options
    - retrying select option action
    - waiting 20ms
    2 × waiting for element to be visible and enabled
      - did not find some options
    - retrying select option action
      - waiting 100ms
    59 × waiting for element to be visible and enabled
       - did not find some options
     - retrying select option action
       - waiting 500ms

2025-06-11 07:18:16,908 - INFO - Closing browser.
2025-06-11 07:18:17 +0000 - dagster - ERROR - __ASSET_JOB - c6d0ef2b-7758-4f37-80ac-1ca033191a6c - 1163 - get_monthly_lotto_result - STEP_FAILURE - Execution of step "get_monthly_lotto_result" failed.

dagster._core.errors.DagsterExecutionStepExecutionError: Error occurred while executing op "get_monthly_lotto_result"::

playwright._impl._errors.Error: Event loop is closed! Is Playwright already stopped?

Stack Trace:
  File "/app/.venv/lib/python3.12/site-packages/dagster/_core/execution/plan/utils.py", line 56, in op_execution_error_boundary
    yield
  File "/app/.venv/lib/python3.12/site-packages/dagster/_utils/__init__.py", line 392, in iterate_with_context
    next_output = next(iterator)
                  ^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/dagster/_core/execution/plan/compute_generator.py", line 129, in _coerce_op_compute_fn_to_iterator
    result = invoke_compute_fn(
             ^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/dagster/_core/execution/plan/compute_generator.py", line 117, in invoke_compute_fn
    return fn(context, **args_to_pass) if context_arg_provided else fn(**args_to_pass)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/src/lotto_project/assets/lotto_result.py", line 178, in get_monthly_lotto_result
    browser.close()
  File "/app/.venv/lib/python3.12/site-packages/playwright/sync_api/_generated.py", line 14264, in close
    return mapping.from_maybe_impl(self._sync(self._impl_obj.close(reason=reason)))
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/.venv/lib/python3.12/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    raise Error("Event loop is closed! Is Playwright already stopped?")

